c7n/resources/sqs.py

```diff
@@ -32,6 +32,11 @@ def _augment(r):
                     QueueUrl=r,
                     AttributeNames=['All'])['Attributes']
                 queue['QueueUrl'] = r
+                if 'queue' in r:
```

Comment:
Hmm, wondering if we might want/need to parse the URL (via [urlparse](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) for example). That might help a couple cases:

* We wouldn't risk entering the URL-tweaking code path for queues that happen to have `queue` in the name
* It may be easier to cover the edge case of `us-east-1`, where the legacy endpoint is `queue.amazonaws.com` with no region baked in

Reply:
@ajkerrigan you raise some great points here, thanks for checking it out. Let me dig in a bit and try to gracefully address these cases.

Reply:
I think a simple check for startswith('https://sqs') should be enough of a conditional check for the reassignment of the queue url.

for the construction we could probably simplify this as well
queue['QueueUrl'] = "https://sqs.{region}.amazonaws.com/{account_id}/{name}".format(
       region=self.manager.config.region, account_id=self.manager.config.account_id, 
       name = r.rsplit('/', 1)[-1])


we also probably need to do better explicit construction of the queue url endpoint for this service client to be compatible with vpc based deployments using vpc endpoints for access, albeit that's distinct from this.